docs/sources/flow/reference/components/discovery.dockerswarm.md

```diff
@@ -0,0 +1,238 @@
+| oauth2              | [oauth2][]        | Configure OAuth2 for authenticating to the endpoint.                               | no       |
+| oauth2 > tls_config | [tls_config][]    | Configure TLS settings for connecting to the endpoint.                             | no       |
+
+The `>` symbol indicates deeper levels of nesting. For example,
+`oauth2 > tls_config` refers to a `tls_config` block defined inside
+an `oauth2` block.
+
+[filter]: #filter-block
+[basic_auth]: #basic_auth-block
+[authorization]: #authorization-block
+[oauth2]: #oauth2-block
+[tls_config]: #tls_config-block
+
+### filter block
+
+Filters can be used to limit the discovery process to a subset of available resources.
+It is possible to define multiple `filter` blocks within the `discovery.dockerswarm` block.
+The list of available filters depends on the `role`:
+
+- [services filters](https://docs.docker.com/engine/api/v1.40/#operation/ServiceList)
+- [tasks filters](https://docs.docker.com/engine/api/v1.40/#operation/TaskList)
+- [nodes filters](https://docs.docker.com/engine/api/v1.40/#operation/NodeList)
+
+The following arguments can be used to configure a filter.
+
+| Name     | Type           | Description                                | Default | Required |
+| -------- | -------------- | ------------------------------------------ | ------- | -------- |
+| `name`   | `string`       | Name of the filter.                        |         | yes      |
+| `values` | `list(string)` | List of values associated with the filter. |         | yes      |
+
+### basic_auth block
+
+{{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" version="<AGENT VERSION>" >}}
+
+### authorization block
+
+{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" version="<AGENT VERSION>" >}}
+
+### oauth2 block
+
+{{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" version="<AGENT VERSION>" >}}
+
+### tls_config block
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" version="<AGENT VERSION>" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+| Name      | Type                | Description                               |
+| --------- | ------------------- | ----------------------------------------- |
+| `targets` | `list(map(string))` | The set of targets discovered from Swarm. |
+
+## Roles
+
+The `role` attribute decides the role of the targets to retrieve.
+
+### services
+
+The `services` `role` discovers all [Swarm services](https://docs.docker.com/engine/swarm/key-concepts/#services-and-tasks) and exposes their ports as targets. For each published port of a service, a single target is generated. If a service has no published ports, a target per service is created using the `port` attribute defined in the arguments.
+
+Available meta labels:
```

Comment:
Future thoughts for a section like this. We are inconsistent with initial caps on the description after the colon. Some topics have inital caps and others don't.  No need to fix this time around... Just something to consider for future improvements in consistency. I would lean towards an initial caps, but either way works.

Reply:
true, we recently added a lot of new components but it should slow down very soon. It will be the opportunity to do some cleanup and to improve the consistency